mlnode/iotdb/mlnode/algorithm/models/NBeats.py

```diff
@@ -0,0 +1,266 @@
+#         backcast_template = torch.cat([backcast_cos_template, backcast_sin_template], dim=0)
+
+#         forecast_cos_template = torch.tensor(np.transpose(np.cos(forecast_grid)), dtype=torch.float32)
+#         forecast_sin_template = torch.tensor(np.transpose(np.sin(forecast_grid)), dtype=torch.float32)
+#         forecast_template = torch.cat([forecast_cos_template, forecast_sin_template], dim=0)
+
+#         self.backcast_basis = nn.Parameter(backcast_template, requires_grad=False)
+#         self.forecast_basis = nn.Parameter(forecast_template, requires_grad=False)
+
+#     def forward(self, theta):
+#         cut_point = self.forecast_basis.shape[0]
+#         backcast = torch.einsum('bp,pt->bt', theta[:, cut_point:], self.backcast_basis)
+#         forecast = torch.einsum('bp,pt->bt', theta[:, :cut_point], self.forecast_basis)
+#         return backcast, forecast
+
+
+class NBeatsBlock(nn.Module):
+    """
+    N-BEATS block which takes a basis function as an argument
+    """
+
+    def __init__(self,
+                 input_size,
+                 theta_size: int,
+                 basis_function: nn.Module,
+                 layers: int,
+                 layer_size: int):
+        """
+        N-BEATS block
+
+        Args:
+            input_size: input sample size
+            theta_size:  number of parameters for the basis function
+            basis_function: basis function which takes the parameters and produces backcast and forecast
+            layers: number of layers
+            layer_size: layer size
+        """
+        super().__init__()
+        self.layers = nn.ModuleList([nn.Linear(in_features=input_size, out_features=layer_size)] +
+                                    [nn.Linear(in_features=layer_size, out_features=layer_size)
+                                     for _ in range(layers - 1)])
+        self.basis_parameters = nn.Linear(in_features=layer_size, out_features=theta_size)
+        self.basis_function = basis_function
+
+    def forward(self, x: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        block_input = x
+        for layer in self.layers:
+            block_input = torch.relu(layer(block_input))
+        basis_parameters = self.basis_parameters(block_input)
+        return self.basis_function(basis_parameters)
+
+
+class NBeatsUnivar(nn.Module):
+    """
+    N-Beats Model.
+    """
+
+    def __init__(self, blocks: nn.ModuleList):
+        super().__init__()
+        self.blocks = blocks
+
+    def forward(self, x):
+        residuals = x
+        forecast = None
+        for _, block in enumerate(self.blocks):
+            backcast, block_forecast = block(residuals)
+            residuals = (residuals - backcast)
+            if forecast is None:
+                forecast = block_forecast
+            else:
+                forecast += block_forecast
+        return forecast
+
+
+block_dict = {
+    'g': GenericBasis,
+    # 't': TrendBasis,
+    # 's': SeasonalityBasis,
```

Comment:
If these will be supported in the future. There is no need to annotate the codes, you can throw an UnsupportException instead.